Controller/SecurityController.php

```diff
@@ -13,33 +12,22 @@ class SecurityController extends Controller
 {
     public function metadataAction()
     {
-        $response = new Response($this->getSPClientInstance()->getSPMetadata());
+        $spclient = $this->get("spid.configuredSPClient");
+        $sp = $spclient->getSpClientConfigurated();
+        $response = new Response($sp->getSPMetadata());
         $response->headers->set('Content-Type', 'xml');
         return $response;
     }
 
-    public function acsAction(Request $request)
-    {
-        $sp = $this->getSPClientInstance();
-
-        if ($sp->isAuthenticated()) {
-            $out = "";
-            foreach ($sp->getAttributes() as $key => $attr) {
-                $out .= $key . ' - ' . $attr . '<br>';
-            }
-            return new Response($out);
-        }
-        throw new \Exception('Unauthenticated');
-    }
-
     public function chooseidpAction(Request $request)
     {
         return new Response('choose one of the configured IDPs <a href="'.$this->container->getParameter('sp_entityid').'/login/local">local</a>');
     }
 
     public function loginAction(Request $request, $idp)
     {
-        $sp = $this->getSPClientInstance();
+        $spclient = $this->get("spid.configuredSPClient");
+        $sp = $spclient->getSpClientConfigurated();
```

Comment:
I don't like this API a lot: you propose to get something that claims to be already configured, then you ask the same exact thing to give you another thing that claims (again) to be configured out of the box.
One possible solution is to rename the service to something closer to what it does right now: wrap the configuration, to provide a convenient handle that lets you retrieve a configured client.
I still wouldn't like it a lot but it would be less misleading for the adopters.
Another possible solution is to have the service class extend the SP class, and at that point having the constructor taking care of the configuration and returning the instance ready for usage.
This is the solution I would prefer, so let's try this road please.